lib/web/resources.go

```diff
@@ -329,14 +343,189 @@ func listResources(clt resourcesAPIGetter, r *http.Request, resourceKind string)
+	}
+
+	resp, err := listResources(clt, r, types.KindAppServer)
+	if err != nil {
+		if trace.IsNotImplemented(err) {
+			appServers, err := clt.GetApplicationServers(r.Context(), apidefaults.Namespace)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+
+			var apps types.Apps
+			for _, server := range appServers {
+				apps = append(apps, server.GetApp())
+			}
+
+			cfg.Apps = types.DeduplicateApps(apps)
+
+			return &listResourcesGetResponse{
+				Items: ui.MakeApps(cfg),
+			}, nil
+		}
+		return nil, trace.Wrap(err)
+	}
+
+	appServers, err := types.ResourcesWithLabels(resp.Resources).AsAppServers()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var apps types.Apps
+	for _, server := range appServers {
+		apps = append(apps, server.GetApp())
+	}
+
+	cfg.Apps = apps
+
+	return &listResourcesGetResponse{
+		Items:      ui.MakeApps(cfg),
+		StartKey:   &resp.NextKey,
+		TotalCount: &resp.TotalCount,
+	}, nil
+}
+
+func handleClusterDesktopsGet(clt resourcesAPIGetter, r *http.Request) (*listResourcesGetResponse, error) {
+	resp, err := listResources(clt, r, types.KindWindowsDesktop)
+	if err != nil {
+		if trace.IsNotImplemented(err) {
+			windowsDesktops, err := clt.GetWindowsDesktops(r.Context(), types.WindowsDesktopFilter{})
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+			windowsDesktops = types.DeduplicateDesktops(windowsDesktops)
+
+			return &listResourcesGetResponse{
+				Items: ui.MakeDesktops(windowsDesktops),
+			}, nil
+		}
+		return nil, trace.Wrap(err)
+	}
+
+	windowsDesktops, err := types.ResourcesWithLabels(resp.Resources).AsWindowsDesktops()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &listResourcesGetResponse{
+		Items:      ui.MakeDesktops(windowsDesktops),
+		StartKey:   &resp.NextKey,
+		TotalCount: &resp.TotalCount,
+	}, nil
+}
+
+func handleClusterKubesGet(clt resourcesAPIGetter, r *http.Request) (*listResourcesGetResponse, error) {
+	resp, err := listResources(clt, r, types.KindKubernetesCluster)
+	if err != nil {
+		if trace.IsNotImplemented(err) {
+			kubeServices, err := clt.GetKubeServices(r.Context())
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+
+			return &listResourcesGetResponse{
+				Items: ui.MakeLegacyKubeClusters(kubeServices),
+			}, nil
+		}
+		return nil, trace.Wrap(err)
+	}
+
+	clusters, err := types.ResourcesWithLabels(resp.Resources).AsKubeClusters()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &listResourcesGetResponse{
+		Items:      ui.MakeKubeClusters(clusters),
+		StartKey:   &resp.NextKey,
+		TotalCount: &resp.TotalCount,
+	}, nil
+}
+
 type listResourcesGetResponse struct {
 	// Items is a list of resources retrieved.
 	Items interface{} `json:"items"`
 	// StartKey is the position to resume search events.
-	StartKey string `json:"startKey"`
+	// Pointer to use the vallue "nil" to determine
+	// if pagination is supported in the web UI.
```

Comment:
```suggestion
	// nil means that pagination is not supported in the Web UI.
```

is that correct?

Same below.